apps/src/tutorialExplorer/filterHeader.jsx

```diff
@@ -0,0 +1,69 @@
+/* FilterHeader: A header that can contain a back button, a "Filter by", a
+ * count of tutorials, and maybe show/hide buttons.
+ */
+
+import React from 'react';
+import BackButton from './backButton';
+
+const FilterHeader = React.createClass({
+  propTypes: {
+    backButton: React.PropTypes.bool,
+    filteredTutorialsCount: React.PropTypes.number.isRequired,
+    mobileLayout: React.PropTypes.bool.isRequired,
+    showingModalFilters: React.PropTypes.bool.isRequired,
+    showModalFilters: React.PropTypes.func.isRequired,
+    hideModalFilters: React.PropTypes.func.isRequired
+  },
+
+  shouldShowOpenFiltersButton() {
+    return this.props.mobileLayout && !this.props.showingModalFilters;
+  },
+
+  shouldShowCloseFiltersButton() {
+    return this.props.mobileLayout && this.props.showingModalFilters;
+  },
+
+  render() {
+    return (
+      <div style={{marginTop: 8, marginBottom: 8}}>
+        {this.props.backButton && (
+          <div>
+            <BackButton/>
+          </div>
+        )}
```

Comment:
Oh, apparently they're needed when the jsx doesn't start on the same line as the return statement.  http://buildwithreact.com/tutorial/components